ocaml/testsuite/tests/typing-modes/modes.ml

```diff
@@ -5,25 +5,25 @@
 (* Let bindings *)
 let local_ foo : string @@ unique = "hello"
 [%%expect{|
-Line 1, characters 4-43:
+Line 1, characters 36-43:
 1 | let local_ foo : string @@ unique = "hello"
-        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+                                        ^^^^^^^
```

Comment:
I still think we should fix this - my reply is at https://github.com/freemagma/flambda-backend/pull/1#discussion_r1659058967

Reply:
Fixed this. Now the modes are always a part of the pattern and expression locs that get created from `vb_pat_constraint` and `vb_expr_constraint`